bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/editor/ClangdConfigurationPage.java

```diff
@@ -93,7 +98,15 @@ private void openRestartDialog() {
 				new String[] { IDialogConstants.NO_LABEL, LspEditorUiMessages.LspEditorPreferencePage_restart_button },
 				1);
 		if (dialog.open() == 1) {
-			LspUtils.getLanguageServers().forEach(w -> w.stop());
+			LspUtils.getLanguageServers().forEach(w -> {
+				try {
+					w.restart();
+				} catch (IOException e) {
+					StatusManager.getManager().handle(
+							new Status(IStatus.ERROR, ClangdPlugin.PLUGIN_ID, "Could not restart language servers"), //$NON-NLS-1$
```

Comment:
Just a minor thing - you can write this more briefly

```suggestion
							Status.error("Could not restart language servers"), //$NON-NLS-1$
```

Reply:
PS - should the `e` be included in the Status so that the full exception can be logged? e.g. 

```java
Status.error("Could not restart language servers", e)  //$NON-NLS-1$
```

Reply:
> PS - should the e be included in the Status so that the full exception can be logged? e.g.

Good point.